src/odemis/driver/smaract.py

```diff
@@ -782,29 +831,36 @@ def moveAbs(self, pos):
     def _estimateMoveDuration(self, new_pos) -> float:
         """
         Estimate the maximum duration of a move
-        new_pos: (dict str -> float): axis name -> absolute target position
+        new_pos: (dict str -> float): axis name -> absolute target position (in user referential)
+        Not all axes have to be provided.
         returns: the duration of the move in seconds
         """
-        pos = self._applyInversion(self.position.value)
-        # TODO: Calculate an estimated move duration
-        # Probably using the speed + accel on the translation could work as a
-        # conservative estimate for translation moves. However for the rotational
-        # moves that's harder. => Just use a hard-coded value for rotations
-        return 30  # s
+        pos = self.position.value
+        # Guess the duration by using the speed and acceleration of the positioners,
+        # for the requested distance.
+        # From Python 3.8, this can be replaced by math.hypot
+        dist_linear = math.sqrt(sum((pos[an] - new_pos[an]) ** 2.0 for an in new_pos.keys() if self.axes[an].unit == "m"))
+        dur_linear = driver.estimateMoveDuration(dist_linear, self._speed, self._accel)
+
+        # Guess the rotation takes maximum 5 s/rad  (it's hard to guess based on the positioners speed)
+        dur_rot =  5 * sum(abs(pos[an] - new_pos[an])for an in new_pos.keys() if self.axes[an].unit == "rad")
+
+        return dur_linear + dur_rot  # s
 
     def _doMoveAbs(self, future, pos):
         """
         Blocking and cancellable absolute move
         future (Future): the future it handles
-        _pos (dict str -> float): axis name -> absolute target position
+        pos (dict str -> float): axis name -> absolute target position, in user referential
         raise:
             SmarPodError: if the controller reported an error
             CancelledError: if cancelled before the end of the move
         """
         last_upd = time.time()
+        pos_hw = self._to_hw_axes(pos)
         dur = self._estimateMoveDuration(pos)
         end = time.time() + dur
-        max_dur = dur * 2 + 1
+        max_dur = dur * 5 + 3
```

Comment:
Could you very briefly explain where the values in dur * 5 + 3 are based on?

Reply:
Done: "The move estimation is really rough, so add some big margin to be on the safe side (we should *never* timeout before a successful move is completed)"